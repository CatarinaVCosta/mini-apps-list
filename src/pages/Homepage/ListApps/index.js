import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { isEmpty } from 'lodash';

/* Components */
import CardComponent from '../../../components/Card/index';

/* Material ui */
import Pagination from '@material-ui/lab/Pagination';
import Alert from '@material-ui/lab/Alert';

/* Containers */
import { getSubscriptions } from '../../../state/fetchApps/src/selectors';

/* Styles */
import './listApps.css';

import PropTypes from 'prop-types';

const propTypes = {
    apps: PropTypes.array,
    subscriptions: PropTypes.object,
    appsFoundInSearch: PropTypes.bool
}

const ERROR_MESSAGE = 'Something went wrong.';

class ListAllApps extends Component {
    constructor(props) {
        super(props);

        this.state = {
            currentPage: 1,
            appsPerPage: 3,
            pageNumbers: 1,
            apps: props.apps
        };
    }

    componentDidMount() {
        const appsToListPerPage = this.getListOfAppsToDisplay()

        this.setState({
            pageNumbers: this.getPageNumbers(appsToListPerPage),
        })
    }

    componentDidUpdate(prevProps) {
        const { apps } = this.props;

        if (prevProps.apps !== apps) {
            const appsToListPerPage = this.getListOfAppsToDisplay()

            this.setState({
                pageNumbers: this.getPageNumbers(appsToListPerPage),
            })
        }
    }

    handleClick = (event, value) => {
        this.setState({
            currentPage: value
        });
    }

    getPageNumbers = appsToListPerPage => {
        const { apps } = this.props;
        let pageNumbersList = [];

        if(!isEmpty(apps)) {
            for (let i = 1; i <= Math.ceil(apps.length / appsToListPerPage.length); i++) {
                pageNumbersList.push(i);
            }
        }

        return pageNumbersList.length;
    }

    getListOfAppsToDisplay = () => {
        const { apps } = this.props;
        const { currentPage, appsPerPage } = this.state;

        if(!isEmpty(apps)) {
            const indexOfLastApp = currentPage * appsPerPage;
            const indexOfFirstApp = indexOfLastApp - appsPerPage;

            return apps.slice(indexOfFirstApp, indexOfLastApp);
        }
    }

    renderSubscriptions = id => {
        const { subscriptions } = this.props;

        return subscriptions[id].map(({name, price}, idx) => 
            <div className='subscriptions' data-test={`app-subscription-${id}`} key={`subscription-${idx}`}>
                <label className='subscriptionName' key={`subscription-name-${id}`}>
                    {name}
                </label>
                <label className='subscriptionPrice' key={`subscription-price-${id}`}>
                    {price}
                </label>
            </div>
        )
    }

    renderCategories = categories => {
        return categories.map(category => 
            <label className='categories'>
                {category}
            </label>
        ).reduce((prev, curr) => [prev, <div className='categories'>/</div>, curr])
    }

    render() {
        const { currentPage, pageNumbers } = this.state;

        const appsToListPerPage = this.getListOfAppsToDisplay();

        return (
            <div className='cardWrapper'>
                {!isEmpty(appsToListPerPage)
                    ? (<Fragment>
                        {appsToListPerPage.map(({id, name, description, categories}) => {
                            const allSubscriptions = this.renderSubscriptions(id);
                            const allCategories = this.renderCategories(categories)
                            return (
                                <div 
                                    key={`app-card-${id}`} 
                                    className='cardComponent' 
                                    data-test={`app-card-${id}`} 
                                >
                                    <CardComponent 
                                        name={name} 
                                        description={description} 
                                        categories={allCategories}
                                        subscriptions={allSubscriptions}
                                    />
                                </div>
                            )
                        })}
                        <Pagination 
                            count={pageNumbers} 
                            shape="rounded" 
                            onChange={this.handleClick} 
                            page={currentPage}
                        />
                    </Fragment>)
                    : <Alert data-test="error-alert" severity="error">{ERROR_MESSAGE}</Alert>
                }
            </div>
        )
    }
}

ListAllApps.propTypes = propTypes;

const mapStateToProps = state => ({
    subscriptions: getSubscriptions(state),
});

export default connect(mapStateToProps)(ListAllApps);