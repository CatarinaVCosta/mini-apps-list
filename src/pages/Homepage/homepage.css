.wrapper {
    display: flex;
    flex-flow: row wrap;
    text-align: center;
    max-width: 1400px;
    margin: 2rem auto;
}

.body {
    flex: 1;
    order: 2;
    text-align: left;
    margin: 2rem; 
}

.title {
    color: black;
    padding: 0 1rem;
}

.menu {
    margin: 2rem; 
    flex: 0;
    order: 1;
    color: darkcyan;
    opacity: 0.7;
}

.loading {
    text-align: center;
    padding-top: 20rem;
}