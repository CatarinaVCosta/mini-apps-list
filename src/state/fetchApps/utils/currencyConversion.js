export const numberFormat = value =>
    new Intl.NumberFormat('de-DE', {
        style: 'currency',
        currency: 'EUR'
    }).format(value);

export const applyPrecision = value => {
    const precision = 2;
    return value/(Math.pow(10, precision)).toFixed(precision);
}
