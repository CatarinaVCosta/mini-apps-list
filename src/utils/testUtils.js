export const state = {
    apps: [
        {
            id: "470fedc5-489e-5acb-a200-c85adaa18356",
            name: "Power Dialer",
            description:
              "Auto dialer that will help increase your connect rates and talk time.",
            categories: ["Dialer"],
            subscriptions: [
                {
                    name: "Trial",
                    price: 0
                },
                {
                    name: "Professional",
                    price: 4500
                },
                {
                    name: "Premium",
                    price: 6000
                }
            ]
        },
        {
            id: "9b565b11-7311-5b5e-a699-97873dffb361",
            name: "Voice Report",
            description: "Calls reporting and analytics of your calls.",
            categories: ["Voice Analytics", "Reporting", "Optimization"],
            subscriptions: [
                {
                    name: "Trial",
                    price: 0
                },
                {
                    name: "Professional",
                    price: 3500
                }
            ]
        }
    ],
    error: '',
    isLoading: false
};

export const sortedCategoriesResponse = ["Dialer", "Optimization", "Reporting", "Voice Analytics"];

export const sortedAppsBySumPlansPriceResponse = [
    {
        id: "9b565b11-7311-5b5e-a699-97873dffb361",
        name: "Voice Report",
        description: "Calls reporting and analytics of your calls.",
        categories: ["Voice Analytics", "Reporting", "Optimization"],
        subscriptions: [
            {
                name: "Trial",
                price: 0
            },
            {
                name: "Professional",
                price: 3500
            }
        ],
        totalSubscriptions: 3500
    },
    {
        id: "470fedc5-489e-5acb-a200-c85adaa18356",
        name: "Power Dialer",
        description:
          "Auto dialer that will help increase your connect rates and talk time.",
        categories: ["Dialer"],
        subscriptions: [
            {
                name: "Trial",
                price: 0
            },
            {
                name: "Professional",
                price: 4500
            },
            {
                name: "Premium",
                price: 6000
            }
        ],
        totalSubscriptions: 10500
    }
];

export const subscriptionsResponse = {
    "9b565b11-7311-5b5e-a699-97873dffb361": [
        {
            name: "Trial",
            price: "Free"
        },
        {
            name: "Professional",
            price: "€ 35.00"
        }
    ],
    "470fedc5-489e-5acb-a200-c85adaa18356": [
        {
            name: "Trial",
            price: "Free"
        },
        {
            name: "Professional",
            price: "€ 45.00"
        },
        {
            name: "Premium",
            price: "€ 60.00"
        }
    ]
}
