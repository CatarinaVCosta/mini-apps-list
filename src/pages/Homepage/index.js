import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { isEmpty, debounce } from 'lodash';

/* Material ui */
import { Divider, CircularProgress, MenuItem } from '@material-ui/core';
import Alert from '@material-ui/lab/Alert';

/* Components */
import MenuComponent from '../../components/Menu/index';
import TextFieldComponent from '../../components/TextField/index';

/* Containers */
import { fetchApps } from '../../state/fetchApps/src/actions';
import { 
    sortedAppsBySumPlansPrice, 
    sortedAppCategories, 
    getIsLoading, 
    getError 
} from '../../state/fetchApps/src/selectors';
import ListAllApps from './ListApps/index';

/* Styles */
import './homepage.css'; 

import PropTypes from 'prop-types';

const SEARCH_LABEL = 'Search by app name';

const propTypes = {
    sortedApps: PropTypes.array,
    isloading: PropTypes.bool,
    error: PropTypes.string,
    sortedAppCategories: PropTypes.array,
}

class HomePage extends Component {
    constructor(props) {
        super(props);

        this.state = {
            filteredAppsByCategory: [],
            filteredAppsByName: [],
        };

        this._filterAppsByCategory = category => this.filterAppsByCategory.bind(this, category);
    }

    componentDidMount() {
        const { dispatch } = this.props;

        dispatch(fetchApps());
    }

    filterAppsByCategory = category => {
        const { sortedApps } = this.props;

        const filteredApps = sortedApps.filter(app => app.categories.includes(category));

        this.setState({ 
            filteredAppsByName: [],
            filteredAppsByCategory: filteredApps,
        })
    }

    searchApp = event => {
        event.preventDefault();

        const { sortedApps } = this.props;
        let filteredApps = [];

        const appName = event.target[0].value;

        if(appName) {
            filteredApps = sortedApps.filter(app => app.name === appName);
        }

        this.setState({ 
            filteredAppsByName: filteredApps,
            filteredAppsByCategory: [],
        })
    }

    appsToList = () => {
        const { filteredAppsByCategory, filteredAppsByName } = this.state;
        const { sortedApps } = this.props;
        
        if(isEmpty(filteredAppsByCategory) && isEmpty(filteredAppsByName)) {
            return sortedApps;
        } else {
            if(isEmpty(filteredAppsByCategory)) {
                return filteredAppsByName;
            } 
            return filteredAppsByCategory;
         }
    }

    onChangeSearchInput = evt => {
        this.debouncedSearch(evt.target.value);
    };

    debouncedSearch = debounce(value => {
        const { sortedApps } = this.props;
        let filteredApps = [];

        filteredApps = sortedApps.filter(app => app.name === value)

        this.setState({ 
            filteredAppsByName: filteredApps,
            filteredAppsByCategory: [],
        })
    }, 500);

    renderLoading = () => {
        return (
            <div data-test="loading" className='loading'>
                <CircularProgress />
            </div>
        )
    }

    renderMenu = () => {
        const { sortedAppCategories } = this.props;

        return(
            <Fragment>
                {!isEmpty(sortedAppCategories) && sortedAppCategories.map((category, index) => {
                    return (
                        <div key={`menu-item-${index}`}>
                            <MenuComponent 
                                item={
                                    <MenuItem 
                                        data-test={`menu-item-category-${index}`}
                                        name={'menuItem'} 
                                        onClick={this._filterAppsByCategory(category)}
                                    >
                                        {category}
                                    </MenuItem>
                                }
                            />
                            <Divider />
                        </div>
                    )
                })}
            </Fragment>
        )        
    }

    render() {
        const { isLoading, error } = this.props;

        return (
            <Fragment>
                {isLoading 
                    ? this.renderLoading()
                    : error
                        ? <Alert data-test="error-alert" severity="error">{error.message}</Alert>
                        : (
                            <div className='wrapper'>
                                <div className='menu'>
                                    <h1 className='title'>Categories</h1>
                                    {this.renderMenu()}
                                </div>
                                <div className='body'>
                                    <form data-test="search-form" >
                                        <TextFieldComponent label={SEARCH_LABEL} onChange={this.onChangeSearchInput}/>
                                        <ListAllApps apps={this.appsToList()} />
                                    </form>
                                </div>
                            </div>
                        )
                }
            </Fragment>
        )
    }
}

HomePage.propTypes = propTypes;

const mapStateToProps = state => ({
    isLoading: getIsLoading(state),
    error: getError(state),
    sortedApps: sortedAppsBySumPlansPrice(state),
    sortedAppCategories: sortedAppCategories(state),
});

export default connect(mapStateToProps)(HomePage);
