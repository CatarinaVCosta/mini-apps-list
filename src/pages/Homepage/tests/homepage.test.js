import { Provider } from 'react-redux';
import { BrowserRouter as Router} from 'react-router-dom';
import { configure, render } from '@testing-library/react';
import React from 'react';
import extend from 'extend';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Homepage from '../index';
import { state } from '../../../utils/testUtils';

configure({ testIdAttribute: 'data-test' });

const storeStateError = extend(true, {}, state);
storeStateError.error = 'error';

const middlewares = [thunk.withExtraArgument({})];
const mockStore = configureMockStore(middlewares);

describe('<Homepage />', () => {
    it('should render correctly', () => {
        // Arrange
        const store = mockStore(state);

        const { container } = render(
            <Router>
                <Provider store={store}>
                    <Homepage />
                </Provider>
            </Router>
        );

        // Assert
        expect(container).toMatchSnapshot();
    });

    it('should render Menu with all app categories', () => {
        // Arrange
        const store = mockStore(state);

        const { getByTestId } = render(
            <Router>
                <Provider store={store}>
                    <Homepage />
                </Provider>
            </Router>
        );

        const firstCategory = getByTestId('menu-item-category-0');
        const secondCategory = getByTestId('menu-item-category-1');
        const thirdCategory = getByTestId('menu-item-category-2');
        const fourthCategory = getByTestId('menu-item-category-3');


        // Assert
        expect(firstCategory).toBeTruthy();
        expect(secondCategory).toBeTruthy();
        expect(thirdCategory).toBeTruthy();
        expect(fourthCategory).toBeTruthy();
    });

    it('should render error message', () => {
        // Arrange
        const store = mockStore(storeStateError);

        const { getByTestId } = render(
            <Router>
                <Provider store={store}>
                    <Homepage />
                </Provider>
            </Router>
        );

        const error = getByTestId('error-alert');

        // Assert
        expect(error).toBeTruthy();
    });
});