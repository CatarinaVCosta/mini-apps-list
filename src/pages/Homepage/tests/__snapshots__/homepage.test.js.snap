// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Homepage /> should render correctly 1`] = `
<div>
  <div
    class="wrapper"
  >
    <div
      class="menu"
    >
      <h1
        class="title"
      >
        Categories
      </h1>
      <div>
        <ul
          class="MuiList-root MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            data-test="menu-item-category-0"
            name="menuItem"
            role="menuitem"
            tabindex="0"
          >
            Dialer
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div>
        <ul
          class="MuiList-root MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            data-test="menu-item-category-1"
            name="menuItem"
            role="menuitem"
            tabindex="0"
          >
            Optimization
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div>
        <ul
          class="MuiList-root MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            data-test="menu-item-category-2"
            name="menuItem"
            role="menuitem"
            tabindex="0"
          >
            Reporting
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
        <hr
          class="MuiDivider-root"
        />
      </div>
      <div>
        <ul
          class="MuiList-root MuiList-padding"
          role="menu"
          tabindex="-1"
        >
          <li
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
            data-test="menu-item-category-3"
            name="menuItem"
            role="menuitem"
            tabindex="0"
          >
            Voice Analytics
            <span
              class="MuiTouchRipple-root"
            />
          </li>
        </ul>
        <hr
          class="MuiDivider-root"
        />
      </div>
    </div>
    <div
      class="body"
    >
      <form
        data-test="search-form"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="text-field"
            id="text-field-label"
          >
            Search by app name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="text-field"
              type="search"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  Search by app name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="cardWrapper"
        >
          <div
            class="cardComponent"
            data-test="app-card-9b565b11-7311-5b5e-a699-97873dffb361"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              classename="card"
            >
              <div
                class="MuiCardContent-root"
              >
                <div
                  class="name"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5"
                  >
                    Voice Report
                  </h5>
                  <p
                    class="MuiTypography-root MuiTypography-body2"
                  >
                    Calls reporting and analytics of your calls.
                  </p>
                  <div
                    class="categories"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2"
                    >
                      <label
                        class="categories"
                      >
                        Voice Analytics
                      </label>
                      <div
                        class="categories"
                      >
                        /
                      </div>
                      <label
                        class="categories"
                      >
                        Reporting
                      </label>
                      <div
                        class="categories"
                      >
                        /
                      </div>
                      <label
                        class="categories"
                      >
                        Optimization
                      </label>
                    </p>
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    <div
                      class="subscriptions"
                      data-test="app-subscription-9b565b11-7311-5b5e-a699-97873dffb361"
                    >
                      <label
                        class="subscriptionName"
                      >
                        Trial
                      </label>
                      <label
                        class="subscriptionPrice"
                      >
                        Free
                      </label>
                    </div>
                    <div
                      class="subscriptions"
                      data-test="app-subscription-9b565b11-7311-5b5e-a699-97873dffb361"
                    >
                      <label
                        class="subscriptionName"
                      >
                        Professional
                      </label>
                      <label
                        class="subscriptionPrice"
                      >
                        € 35.00
                      </label>
                    </div>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="cardComponent"
            data-test="app-card-470fedc5-489e-5acb-a200-c85adaa18356"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
              classename="card"
            >
              <div
                class="MuiCardContent-root"
              >
                <div
                  class="name"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5"
                  >
                    Power Dialer
                  </h5>
                  <p
                    class="MuiTypography-root MuiTypography-body2"
                  >
                    Auto dialer that will help increase your connect rates and talk time.
                  </p>
                  <div
                    class="categories"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2"
                    >
                      <label
                        class="categories"
                      >
                        Dialer
                      </label>
                    </p>
                  </div>
                  <p
                    class="MuiTypography-root MuiTypography-body1"
                  >
                    <div
                      class="subscriptions"
                      data-test="app-subscription-470fedc5-489e-5acb-a200-c85adaa18356"
                    >
                      <label
                        class="subscriptionName"
                      >
                        Trial
                      </label>
                      <label
                        class="subscriptionPrice"
                      >
                        Free
                      </label>
                    </div>
                    <div
                      class="subscriptions"
                      data-test="app-subscription-470fedc5-489e-5acb-a200-c85adaa18356"
                    >
                      <label
                        class="subscriptionName"
                      >
                        Professional
                      </label>
                      <label
                        class="subscriptionPrice"
                      >
                        € 45.00
                      </label>
                    </div>
                    <div
                      class="subscriptions"
                      data-test="app-subscription-470fedc5-489e-5acb-a200-c85adaa18356"
                    >
                      <label
                        class="subscriptionName"
                      >
                        Premium
                      </label>
                      <label
                        class="subscriptionPrice"
                      >
                        € 60.00
                      </label>
                    </div>
                  </p>
                </div>
              </div>
            </div>
          </div>
          <nav
            aria-label="pagination navigation"
            class="MuiPagination-root"
          >
            <ul
              class="MuiPagination-ul"
            >
              <li>
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                    />
                  </svg>
                </button>
              </li>
              <li>
                <button
                  aria-current="true"
                  aria-label="page 1"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded Mui-selected"
                  tabindex="0"
                  type="button"
                >
                  1
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </li>
              <li>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded Mui-disabled Mui-disabled"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiPaginationItem-icon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </form>
    </div>
  </div>
</div>
`;
